sphinx_gallery/gen_rst.py

```diff
@@ -498,11 +498,8 @@ def generate_file_rst(fname, target_dir, src_dir, gallery_conf):
     ref_fname = example_file.replace(os.path.sep, '_')
     example_rst = """\n\n.. _sphx_glr_{0}:\n\n""".format(ref_fname)
 
-    if not fname.startswith('plot'):
-        convert_func = dict(code=codestr2rst, text=text2string)
-        for blabel, bcontent in script_blocks:
-            example_rst += convert_func[blabel](bcontent) + '\n'
-    else:
+    filename_pattern = gallery_conf.get('filename_pattern', 'plot_*')
+    if fnmatch.fnmatch(fname, filename_pattern):
```

Comment:
> I am in favour of using re.search on the full example path rather than
> just the basename this way you can filter by directories (which is
> what @jasmainak use case was originally IIRC). If you think plot_' is
> too liberal then just use/plot_as default (probably need to use os.sep
> or(\|/)plot_` for Windows though)

Ok to include directories matching is just about to match against
`src_file` and not `fname`, so easy. I'm not sure if os.sep or
the slashes will become an issue, as for the re string slashed have to
be escaped. Taking the liberal `plot` as default can't be too bad,
saying if the file has plot written at any place it will build by default.